src/saltext/vmware/utils/license_mgr.py

```diff
@@ -0,0 +1,448 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+# pylint: disable=no-name-in-module
+import logging
+
+# noreorder
+import salt.exceptions
+import salt.modules.cmdmod
+import salt.utils.path
+import salt.utils.platform
+import salt.utils.stringutils
+import saltext.vmware.utils.cluster as utils_cluster
+import saltext.vmware.utils.datacenter as utils_datacenter
+import saltext.vmware.utils.esxi as utils_esxi
+from saltext.vmware.utils.common import get_service_content as get_service_content
+
+try:
+    from pyVmomi import vim, vmodl
+
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+log = logging.getLogger(__name__)
+
+
+def _list_lic_keys(licenses):
+    """
+    Return a list of license keys
+
+    licenses:
+       vCenter licenses as made available from licenseManager
+
+    Note:
+        skips evaluation licenses
+    """
+    log.debug("started _list_lic_keys")
+    lic_keys = []
+    for lic in licenses:
+        if lic.used is None:
+            continue
+        lic_keys.append(lic.licenseKey)
+    return lic_keys
+
+
+def _find_lic_for_key(licenses, license_key):
+    """
+    Return license which matches specified license_key
+
+    licenses:
+       vCenter licenses as made available from licenseManager
+
+    license_key
+        license key to search for
+
+    Note:
+        skips evaluation licenses
+    """
+    log.debug("started _find_lic_for_key")
+    lic_keys = []
+    for lic in licenses:
+        if lic.used is None:
+            continue
+        if license_key == lic.licenseKey:
+            return lic
+    return None
+
+
+def is_vcenter(service_instance):
+    """
+    Test if service_instance represents vCenter,
+    otherwise assume represents an ESXi server
+
+    service_instance
+        The Service Instance to check if it represents a vCenter
+
+    Return:
+        True  - vCenter
+        False - ESXi server
+        None - neither of the above
+    """
+    try:
+        srv_content = get_service_content(service_instance)
+        apitype = srv_content.about.apiType
+    except vim.fault.VimFault as exc:
+        log.exception(exc)
+        raise salt.exceptions.VMwareApiError(exc.msg)
+    except vmodl.RuntimeFault as exc:
+        log.exception(exc)
+        raise salt.exceptions.VMwareRuntimeError(exc.msg)
+    except AttributeError as exc:
+        log.exception(exc)
+        raise salt.exceptions.CommandExecutionError(exc.msg)
+    except TypeError as exc:
+        log.exception(exc)
+        raise salt.exceptions.CommandExecutionError(exc.msg)
+
+    if apitype == "VirtualCenter":
+        return True
+    elif apitype == "HostAgent":
+        return False
+
+    return None
+
+
+def get_license_mgr(service_instance):
+    """
+    Return a license manager from specified Service Instance
```

Comment:
Just my style, like assigning to variables which I can then easily view in a debugger.
Compilers optimize it out with -O2 for release builds.
Did I mention I was an old strongly-typed kind of guy :D , and optimized Python ROFAL (actually a think nowadays not like 1.x, and I really did like PyPy back in 2.4 usage).

Always like to see what gets returned, Can change if strongly object.

Reply:
Changed to your suggestion, Python - clearest possible, not those nasty C/C++ habits :D

Reply:
I just meant that the above text was more accurate to what the code is doing :joy: 

Specifically, the license manager is only returned if the SI is connected to vCenter (line 110-111 returns None if it's connected to anything else)

Reply:
To be entirely clear:

```suggestion
    Return a license manager from specified Service Instance if the Service Instance is connected to vCenter, otherwise return None.
```

Reply:
done